cmd/argocd-application-controller/main.go

```diff
@@ -109,6 +111,7 @@ func newCommand() *cobra.Command {
 	command.Flags().IntVar(&glogLevel, "gloglevel", 0, "Set the glog logging level")
 	command.Flags().IntVar(&metricsPort, "metrics-port", common.DefaultPortArgoCDMetrics, "Start metrics server on given port")
 	command.Flags().IntVar(&selfHealTimeoutSeconds, "self-heal-timeout-seconds", 5, "Specifies timeout between application self heal attempts")
+	command.Flags().Int64Var(&kubectlParallelismLimit, "kubectl-parallelism-limit", 0, "Number of allowed concurrent kubectl fork/execs.")
```

Comment:
Would it be a good idea to have a sensible default? And make this mandatory, not optional.

Reply:
It is difficult to find default without testing in production like environment. I was hoping to backport it into 1.2.1, deploy collect data and then send PR with the default and documentation.